### ISSUE TITLE
PRs that have passed CI, but have "aged" considerably w.r.t. master, can still be merged.

### ISSUE THREAD

**Issue by DickJC123** [apache/incubator-mxnet#18753]
## Description
Could our system be changed to prevent merging of a CI-passing PR after a "merge window" has elapsed?  I guess this is not normally a problem since after a time an auto-merge would fail and the PR becomes unmergable, right?

PR https://github.com/apache/incubator-mxnet/pull/18197, merged today having passed CI 2 months ago, becomes the poster-child for this issue.  Thanks @leezu for submitting reverting PR https://github.com/apache/incubator-mxnet/pull/18750.

**Comment by marcoabreu** [comment]
Agree, that should be easy to solve by having a lambda function triggered nightly which invalidates status checks which are older than a certain time period.

@ChaiBapchya

**Comment by ChaiBapchya** [comment]
Sure that should be doable. And what would be that time period? 2/4/8 weeks?

**Comment by marcoabreu** [comment]
Let's start with 2 weeks - as long as it's configurable :)

**Comment by DickJC123** [comment]
When my in-development PRs become unmergeable due to other accepted PRs, I merge the current master with my PR and push the result, which generates another CI run.  Is it possible that the PR I mentioned above, after 2 months was still mergeable?  Or is there some other way to fix merge conflicts in a way that doesn't require a repeated CI run?
